Sources/KlaviyoSwift/StateManagement.swift

```diff
@@ -92,7 +92,11 @@ enum KlaviyoAction: Equatable {
 
     var requiresInitialization: Bool {
         switch self {
-        case .setEmail, .setPhoneNumber, .setExternalId, .setPushToken, .enqueueEvent, .enqueueProfile, .setProfileProperty, .resetProfile, .resetStateAndDequeue:
+        case let .enqueueEvent(event):
+            // if event metric is opened push we DON'T require initilization in all other event metric cases we DO.
+            return event.metric.name == .OpenedPush ? false : true
```

Comment:
this can be simplified to `return event.metric.name != .OpenedPush` but felt that was not readable.

Reply:
hmm to me that seems more readable maybe

Reply:
I'm assuming syntax doesn't allow for something like `case let .enqueueEvent(event) && event.metric.name .OpenedPush :` right? 
Else i agree with Noah on readability

Reply:
Yeah within a switch that syntax AFAIK isn't possible. Updated to `return event.metric.name != .OpenedPush`. I think I can get used to that.

Reply:
Oh yah this might be possible one sec...

Reply:
```swift
    var requiresInitialization: Bool {
        switch self {
        case let .enqueueEvent(event) where event.metric.name == .OpenedPush:
            // if event metric is opened push we DON'T require initilization in all other event metric cases we DO.
            return false
        
        case .enqueueEvent:
            // if event metric is opened push we DON'T require initilization in all other event metric cases we DO.
            return true

        case .setEmail, .setPhoneNumber, .setExternalId, .setPushToken, .enqueueProfile, .setProfileProperty, .resetProfile, .resetStateAndDequeue:
            return true

        case .initialize, .completeInitialization, .deQueueCompletedResults, .networkConnectivityChanged, .flushQueue, .sendRequest, .stop, .start, .cancelInFlightRequests, .requestFailed:
            return false
        }
    }
```

Something like this?

Reply:
May be can combine the`enqueueEvent ` with the rest of the cases that require initilization

Reply:
```swift
    var requiresInitialization: Bool {
        switch self {
        case let .enqueueEvent(event) where event.metric.name == .OpenedPush:
            // if event metric is opened push we DON'T require initilization in all other event metric cases we DO.
            return false
        
        case .setEmail, .setPhoneNumber, .setExternalId, .setPushToken, .enqueueProfile, .setProfileProperty, .resetProfile, .resetStateAndDequeue, .enqueueEvent:
            return true

        case .initialize, .completeInitialization, .deQueueCompletedResults, .networkConnectivityChanged, .flushQueue, .sendRequest, .stop, .start, .cancelInFlightRequests, .requestFailed:
            return false
        }
    }
```

Reply:
yah I like that! forgot about where. yah and move enqueueEvent under the other true case.

Reply:
Done!

Reply:
Using `where` now? Maybe github is messed up, but i'm not seeing that change